chain/client/src/client.rs

```diff
@@ -2210,34 +2210,37 @@ impl Client {
         // here it is fine to use `cur_block_header` as it is a best effort estimate. If the transaction
         // were to be included, the block that the chunk points to will have height >= height of
         // `cur_block_header`.
-        if let Err(e) = self
-            .chain
-            .chain_store()
-            .check_transaction_validity_period(&cur_block_header, tx.transaction.block_hash())
-        {
-            debug!(target: "client", ?tx, "Invalid tx: expired or from a different fork");
+        if let Err(e) = self.chain.chain_store().check_transaction_validity_period(
+            &cur_block_header,
+            signed_tx.transaction.block_hash(),
+        ) {
+            debug!(target: "client", ?signed_tx, "Invalid tx: expired or from a different fork");
             return Ok(ProcessTxResponse::InvalidTx(e));
         }
         let gas_price = cur_block_header.next_gas_price();
         let epoch_id = self.epoch_manager.get_epoch_id_from_prev_block(&head.last_block_hash)?;
         let protocol_version = self.epoch_manager.get_epoch_protocol_version(&epoch_id)?;
         let shard_layout =
             self.epoch_manager.get_shard_layout_from_protocol_version(protocol_version);
-        let receiver_shard = shard_layout.account_id_to_shard_id(tx.transaction.receiver_id());
+        let receiver_shard =
+            shard_layout.account_id_to_shard_id(signed_tx.transaction.receiver_id());
         let receiver_congestion_info =
             cur_block.block_congestion_info().get(&receiver_shard).copied();
 
-        if let Err(err) = self.runtime_adapter.validate_tx(
+        let validated_tx = match self.runtime_adapter.validate_tx(
             &shard_layout,
-            tx,
+            signed_tx.clone(),
             protocol_version,
             receiver_congestion_info,
         ) {
```

Comment:
Style (not necessary to act on this but something to keep in mind for the future): I find that putting the scrutinee into its own binding like this:

```
let scrutinee = self.runtime_adapter.validate_tx(
            &shard_layout,
            signed_tx.clone(),
            protocol_version,
            receiver_congestion_info,
        );
let result = match scrutinee {
    ...
};
``` 

tends to format to code that's both more compact and easier to read.

Reply:
good point, will keep it in mind for future.